src/main.cpp

```diff
@@ -6034,3 +6034,44 @@ bool IsResearcher(const std::string& cpid)
 {
     return cpid.length() == 32;
 }
+
+/** Fees collected in block by miner **/
+int64_t GetFeesCollected(const CBlock& block)
+{
+    // Return 0 if there is no transactions in block.
+    if (block.vtx.size() < 2)
+        return 0;
+
+    int64_t nFees = 0;
+
+    for (unsigned int i = 2; i < block.vtx.size(); i++)
+    {
+        int64_t nDebit = 0;
+        int64_t nCredit = 0;
+        const CTransaction& txData = block.vtx[i];
+
+        // Scan inputs
+        for (const auto& txvinDataParse : txData.vin)
+        {
+            uint256 hashblock;
+            CTransaction txvinData;
+            const COutPoint& txinputData = txvinDataParse.prevout;
+
+            if (GetTransaction(txinputData.hash, txvinData, hashblock))
```

Comment:
Using the `GetTransaction()` function here is very expensive because it creates and destroys `CTxDB` handles on each iteration, and it reads the blocks from disk and hashes them even though we don't need that data. This can slow down services like explorers and exchanges that use the `getblock` RPC interface to fetch much of their data from the wallet. Instead, we can load the transaction directly:

```cpp
CTxDB txdb("r"); // outside the loop 
...
CTransaction txvinData;

if (!txvinData.ReadFromDisk(txdb, txinputDataParse.prevout)) {
    return 0;
}
```

Reply:
ill find another way :)

Reply:
looks like all i can see is this way but can do a little less then gettransaction for similar result using GetvalueIn which uses fetch inputs and still reads from txdb

Reply:
but i can do it in bulk so its done once

Reply:
Using the txdb is fine--we need to load the transaction. I'm talking about how `GetTransaction()` reads the entire block from disk. Did you try the suggestion that I posted above?

Reply:
ya i did what u did above similar i chose to use ReadDiskTx.